server/mergin/sync/public_api_controller.py

```diff
@@ -177,12 +178,15 @@ def add_project(namespace):  # noqa: E501
 
     :rtype: None
     """
-    request.json["name"] = request.json["name"].strip()
+    name = request.json["name"].strip()
 
-    if not is_name_allowed(request.json["name"]):
+    if not name:
```

Comment:
I would put .strip() and empty check into project_name_validation function to make sure it is applied everywhere as it is generic prerequisite.